lib/3scale_toolbox/cli/error_handler.rb

```diff
@@ -0,0 +1,133 @@
+module ThreeScaleToolbox::CLI
+  # Catches errors and prints nice diagnostic messages
+  class ErrorHandler
+    def self.error_watchdog(exit_on_error: true)
+      new.error_watchdog(exit_on_error: exit_on_error) { yield }
+    end
+
+    def error_watchdog(exit_on_error:)
+      # Set exit handler
+      %w[INT TERM].each do |signal|
+        Signal.trap(signal) do
+          puts
+          exit!(0)
+        end
+      end
+
+      # Set stack trace dump handler
+      if !defined?(RUBY_ENGINE) || RUBY_ENGINE != 'jruby'
+        begin
+          Signal.trap('USR1') do
+            puts 'Caught USR1; dumping a stack trace'
+            puts caller.map { |i| "  #{i}" }.join("\n")
+          end
+        rescue ArgumentError
```

Comment:
Rubocop should scream about this. Why this error can happen? At least a comment explaining why it would be useful.